src/fu-device-list.c

```diff
@@ -554,6 +554,14 @@ fu_device_list_item_finalized_cb(gpointer data, GObject *where_the_object_was)
 static void
 fu_device_list_item_set_device(FuDeviceItem *item, FuDevice *device)
 {
+	/* TODO: move the progress to the write_firmware vfunc and break ABI */
```

Comment:
this one too, I think should just be done now.  The folks working on plugins can easily be helped when they submit their PR or rebase on master.

Reply:
I've done this. Phew!